core/src/Cuda/Kokkos_Cuda_GraphNodeKernel.hpp

```diff
@@ -34,6 +34,25 @@
 namespace Kokkos {
 namespace Impl {
 
+template <typename Functor>
+struct GraphNodeThenHostImpl<Kokkos::Cuda, Functor> {
+  Functor m_functor;
+  cudaGraphNode_t m_node = nullptr;
```

Comment:
I'd like to see a constructor here in particular since a class instance since there is no direct constructor call. Hence, I find it hard to understand how the class is initialized if there is no constructor.

Reply:
I rather disallow manual initializations of internal members rather than only discourage. Restricting what the constructor is initializing and what not, improves clarity.

Reply:
Done :checkered_flag: